### ISSUE TITLE
Add links to blurbs when clicking underlined text

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/site-www#6]
Especially in the initial code example, we should link to docs explaining each feature that is highlighted. For instance if you click `await` in the `await for (var estimate in computePi()) {` line, you get this text `Language-level asynchrony support makes you more productive when dealing with deeply asynchronous code (like most UI code, for example).`. If you have never seen async/await before, I don't think that is going to help you understand what is actually going on. A simple link to somewhere describing the feature would help a lot. This applies most to features like async/await and generators, which might not be well known to a lot of devs.

**Comment by filiph** [comment]
Great point. Will do this.

**Comment by kwalrath** [comment]
I'd still like to do this, but doubt Filip will get the time.

**Comment by chalin** [comment]
The problem is that the current "tool tips" are plain text. While Kathy was away I prototyped an implementation that supports HTML tool tips.  Would you like me to complete that implementation? @filiph, if you'd prefer doing it let me know.

**Comment by filiph** [comment]
@chalin Please go ahead with your implementation. I was thinking of the minimal one, where each underlined link could have zero or one links attached to it as a `data-*` attribute. This would mean that you could only have things like:

> Language-level asynchrony support makes you more productive when dealing with deeply asynchronous code (like most UI code, for example). [Link.](https://www.dartlang.org/articles/language/await-async)

In other words, it wouldn't be able to support full HTML. The benefit is simplicity.

Just my 2 cents. I won't have time to really implement this in the foreseeable future. Since you have something prototyped already, let's go ahead with that.

**Comment by chalin** [comment]
FYI, tooltips supporting HTML are in PR #526.

@kwalrath, et all: what kind of link prose did you have in mind? Although there is no limit to the amount text we can add to the tooltips, I'm assuming that we'll still want to be terse.

Maybe just end the tip with: [Read more.](#)

I assume that we'll want such a link to pretty much every entry, and that the link should be to the appropriate entry of the Language Tour?

Once we have a general agreement on the kinds of links we want, and once #526 is merged, I'll put something together.

**Comment by kwalrath** [comment]
Read more sgtm

**Comment by anders-sandholm** [comment]
I like the look of the new "popup". I observed two (related) issues:
1) On desktop they appear as soon as I hover. Previous behavior was that I had to click before anything appeared. I prefer having to click to toggle whether the popup gets shown.
2) If we support HTML in order to be able to include links, the current desktop behavior of "show on hover and disappear again as soon as you move the mouse" makes it impossible to ever click the link. 

LMK if I should file a new issue for this

**Comment by chalin** [comment]
I'll address that as part of this issue. Thanks for the feedback.

**Comment by chalin** [comment]
@anders-sandholm - actually, #530 is a new issue covering the point you raised along with another problem I identified.

**Comment by kwalrath** [comment]
This is no longer blocked by the infrastructure, but it's difficult to make changes to the sample right now, and a bunch of site (infra)structure changes might happen soon.

@mit-mit, how would you like us to track issues like this where I'm reluctant to implement them now, for infrastructure/structure reasons? Assign them to a project? Add a label?

**Comment by kwalrath** [comment]
Actually, with @chalin's changes in #1303, it should be easier to implement this.

**Comment by parlough** [comment]
Closing this, as we will likely replace this functionality (https://github.com/dart-lang/site-www/issues/2316) instead.
